fix(export): add id of rollouts/constraints to export

crossy-l opened the pull request:
In https://github.com/orgs/flipt-io/discussions/4311, the audit logs were already improved. Now this is a small follow up.
I want to add the id's of the rollouts/segments to the flipt export. This way, when an audit log comes in, the previous version of that record can easily be looked up using it's id in a previous export.

This PR simply adds the ids to the export interfaces.

The tests still need to be updated however. That is of course, if you agree with the changes.

_Also this is my first ever open source PR :)_

crossy-l committed:
fix(export): add id of rollouts/constraints to export

Signed-off-by: crossy-l <crossyl2001@gmail.com>

markphelps reviewed (commented):
thanks @crossy-l , looks good to me to continue

I think we should also bump the version here to 1.5 too , even though its a nonbreaking change https://github.com/flipt-io/flipt/blob/af21da60613fe300344aabca736328e88db6a9ab/internal/ext/exporter.go#L23

crossy-l commented:
Thanks. I will look into the test cases and bump the version in the following days.

crossy-l committed:
chore(export): Overwrite existing fixtures with auto generated ones

Signed-off-by: crossy-l <crossyl2001@gmail.com>

crossy-l committed:
Merge branch 'flipt-io:main' into lk/fix-flipt-export-ids

markphelps commented:
@crossy-l dont worry about the formatting I'll run our formatter `prettier` on those files

markphelps committed:
Merge branch 'main' into lk/fix-flipt-export-ids

markphelps committed:
Merge branch 'main' into lk/fix-flipt-export-ids

markphelps committed:
Merge branch 'main' into lk/fix-flipt-export-ids

markphelps committed:
Merge branch 'main' into lk/fix-flipt-export-ids

markphelps committed:
Merge branch 'main' into lk/fix-flipt-export-ids

kodiakhq[bot] committed:
Merge branch 'main' into lk/fix-flipt-export-ids

kodiakhq[bot] committed:
Merge branch 'main' into lk/fix-flipt-export-ids

kodiakhq[bot] committed:
Merge branch 'main' into lk/fix-flipt-export-ids

kodiakhq[bot] committed:
Merge branch 'main' into lk/fix-flipt-export-ids

kodiakhq[bot] committed:
Merge branch 'main' into lk/fix-flipt-export-ids

kodiakhq[bot] committed:
Merge branch 'main' into lk/fix-flipt-export-ids

Commit:
fix: update tests to handle ID fields in export

The PR #4359 adds ID fields to rollouts and constraints in the export.
This change updates the tests to handle these new fields properly:

1. CLI tests now check for individual fields rather than exact YAML blocks
   since IDs are included between the fields
2. Integration tests strip ID fields using regex before comparison
   since IDs are server-generated UUIDs and not stable across import/export

This preserves the PR's intent (adding IDs to exports) while fixing tests.

Signed-off-by: markphelps <mark.phelps@outlook.com>
